renderdoc/driver/vulkan/wrappers/vk_resource_funcs.cpp

```diff
@@ -1525,6 +1525,9 @@ VkResult WrappedVulkan::vkBindBufferMemory(VkDevice device, VkBuffer buffer, VkD
     // memory that has been allocated but not used, but that will be skipped or postponed as
     // appropriate.
     GetResourceManager()->MarkDirtyResource(GetResID(memory));
+
+    if(m_DeviceAddressResources.IDs.contains(record->GetResourceID()))
```

Comment:
This access needs to lock around the lookup or else it could be racing against an ID being inserted during creation/allocation on another thread.